metadata-db-app/int_test/cmr/metadata_db/int_test/concepts/variable_save_test.clj

```diff
@@ -7,20 +7,96 @@
+  respective revision ids).
+
+  The results of this function are intended to be used with `(apply ...)`."
+  [concept-id initial-revision-id second-revision-id tombstone-revision-id
+   final-revision-id]
+  (mapv #(:concept (util/get-concept-by-id-and-revision concept-id %))
+        [initial-revision-id second-revision-id tombstone-revision-id
+         final-revision-id]))
 
 ;; tests
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-(deftest save-variable-test
-  (c-spec/general-save-concept-test :variable ["REG_PROV"]))
+(deftest save-variable
+  (c-spec/general-save-concept-test :variable ["PROV1"]))
 
-(deftest save-variable-with-missing-required-parameters-test
+(deftest save-variable-with-missing-required-parameters
   (c-spec/save-test-with-missing-required-parameters
-    :variable ["REG_PROV"] [:concept-type :provider-id :native-id :extra-fields]))
+    :variable ["PROV1"] [:concept-type :provider-id :native-id :extra-fields]))
+
+(deftest save-variable-created-at
+  (testing "Save variable multiple times gets same created-at"
+    (let [initial-var (util/variable-concept "PROV1" 2)
+          ;; 1) Save a variable
+          ;; 2) Then wait for a small period of time before saving it again
+          ;; 3) Then wait again and save a tombstone.
+          ;; 4) Finally, wait a bit and save a new (non-tombstone) revision.
+          ;;
+          ;; All should have the same `created-at` value.
+          ;;
+          ;; Note - Originally planned to use the time-keeper functionality
+          ;; for this, but metdata-db tests don't have access to the control
+          ;; api that would allow this to work in CI.
+          {concept-id :concept-id
+           initial-revision-id :revision-id} (util/save-concept initial-var)
+          _ (Thread/sleep 100)
+          {second-revision-id :revision-id} (util/save-concept initial-var)
+          _ (Thread/sleep 100)
+          {tombstone-revision-id :revision-id} (util/delete-concept
+                                                concept-id)
+          _ (Thread/sleep 100)
+          {final-revision-id :revision-id} (util/save-concept initial-var)
+          revisions (get-revisions concept-id initial-revision-id
+                                   second-revision-id tombstone-revision-id
+                                   final-revision-id)]
+      (is (apply util/created-at-same? revisions)))))
+
+(deftest save-variable-with-conflicting-native-id
+  (testing "Save a variable"
+    (let [var1 (util/variable-concept "PROV1" 1)
+          {:keys [status revision-id concept-id]} (util/save-concept var1)]
+      (is (= status 201))
+      (is (= 1 revision-id))
+      (testing "and another with all the same data"
+        (let [var2 var1
+              {:keys [status revision-id]} (util/save-concept var1)]
+          (is (= 201 status))
+          (is (= 2 revision-id))))
+      (testing "and another with same data but different provider"
+        (let [var3 (util/variable-concept "PROV2" 1)
```

Comment:
Change this to `(let [var3 (assoc var1 :provider-id "PROV2")` to make it clearer what is different.